### PR TITLE
Wrap DB exceptions in Transformers

### PR DESCRIPTION
**Pull request by MichaelA-ops**

### CODE REVIEW

File: src/main/java/cloud/filibuster/instrumentation/libraries/dynamic/proxy/DynamicProxyInterceptor.java [hunk 1]
```diff
@@ -172,6 +173,16 @@ public Object invoke(Object proxy, Method method, Object[] args) throws Throwabl
         String futureInvokeExceptionOnMethod = null;
         JSONObject futureExceptionMetadata = null;
 
+        if (transformerFault != null && filibusterClientInstrumentor.shouldAbort()) {
+            Object transformerFaultValue = getTransformerFaultValue(filibusterClientInstrumentor, transformerFault, method.getReturnType());
+
+            if (transformerFaultValue instanceof DBException) {
```
Comment (cmeiklejohn) [inline src/main/java/cloud/filibuster/instrumentation/libraries/dynamic/proxy/DynamicProxyInterceptor.java:179]: What does DB Exception represent?

Comment (MichaelA-ops) [reply src/main/java/cloud/filibuster/instrumentation/libraries/dynamic/proxy/DynamicProxyInterceptor.java:179]: It's a class representing DB exceptions that I wrap in transformers.

Comment (MichaelA-ops) [reply src/main/java/cloud/filibuster/instrumentation/libraries/dynamic/proxy/DynamicProxyInterceptor.java:179]: I could just use a JSONObject instead, but using a class with its own builder makes it more organised

**Review by cmeiklejohn — CHANGES_REQUESTED** [review]
see comments

File: src/main/java/cloud/filibuster/instrumentation/libraries/dynamic/proxy/DynamicProxyInterceptor.java (hunk 1 above)
Comment (MichaelA-ops) [reply src/main/java/cloud/filibuster/instrumentation/libraries/dynamic/proxy/DynamicProxyInterceptor.java:179]: It shouldn't make a difference. Exception in this PR are wrapped in the class `DBExceptionTransformer`. The result of this transformation is a `DBException`, regardless of whether we write to a JSON file and read it back in.

Comment (MichaelA-ops) [reply src/main/java/cloud/filibuster/instrumentation/libraries/dynamic/proxy/DynamicProxyInterceptor.java:179]: Just to be absolutely sure though, how do I make Filibuster write to a JSON file and read back in?

Comment (cmeiklejohn) [reply src/main/java/cloud/filibuster/instrumentation/libraries/dynamic/proxy/DynamicProxyInterceptor.java:176]: There is a function in the configuration file super type that contains the code to write the file, you use the annotation and point it at the file to read it back in.

**Review by cmeiklejohn — CHANGES_REQUESTED** [review]
.

File: src/main/java/cloud/filibuster/instrumentation/libraries/dynamic/proxy/DynamicProxyInterceptor.java (hunk 1 above)
Comment (MichaelA-ops) [reply src/main/java/cloud/filibuster/instrumentation/libraries/dynamic/proxy/DynamicProxyInterceptor.java:176]: I have just added a test where I am importing a `.fac` file containing a transformer-wrapped exception. The test is successful.

Comment (MichaelA-ops) [reply src/main/java/cloud/filibuster/instrumentation/libraries/dynamic/proxy/DynamicProxyInterceptor.java:176]: The file name is `RedisFilibusterTransformerExceptionWithAnalysisResourceFileTest`

**Pull request merged**